src/main/kotlin/no/nav/sbl/sosialhjelp_mock_alt/integrations/azure/AzureController.kt

```diff
@@ -48,45 +49,63 @@ class AzureController(val pdlService: PdlService) {
         }
     }
 
+    @GetMapping("/azuread/graph/me/memberOf")
+    fun getInloggetAzureBrukersGrupper(@RequestHeader headers: HttpHeaders): ResponseEntity<AzureAdGrupper> {
+        val id = hentFnrFraToken(headers)
+        log.info("Henter azureAd brukers grupper for \"me\" (id: $id)")
+        return try {
+            val grupper = hentGrupper(id)
+            ResponseEntity.ok(AzureAdGrupper(value = grupper))
+        } catch (e: MockAltException) {
+            log.info("Feil ved henting av brukers grupper: ${e.message}")
+            ResponseEntity.status(HttpStatus.UNAUTHORIZED).build()
+        }
+    }
+
     @GetMapping("/azuread/graph/users/{id}/memberOf")
     fun getAzureBrukersGrupper(@PathVariable id: String): ResponseEntity<AzureAdGrupper> {
         log.info("Henter azureAd brukers grupper for brukerId $id")
         return try {
-            val personalia = pdlService.getPersonalia(id)
-            val grupper = if (personalia.navn.fornavn == "Admin") {
-                if (personalia.navn.etternavn == "Admin") {
-                    listOf(
-                        AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-veileder", "Veiledere"),
-                        AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-admin", "Administratorer"),
-                    )
-                } else if (personalia.navn.etternavn == "Arkiv") {
-                    listOf(
-                        AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-arkiv", "Teknisk-arkiv"),
-                    )
-                } else if (personalia.navn.etternavn == "Innsikt") {
-                    listOf(
-                        AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-innsikt", "Innsikktere"),
-                    )
-                } else {
-                    listOf(
-                        AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-veileder", "Veiledere"),
-                        AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-admin", "Administratorer"),
-                        AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-arkiv", "Teknisk-arkiv"),
-                        AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-innsikt", "Innsikktere"),
-                    )
-                }
-            } else if (personalia.navn.fornavn == "Tyske") {
-                emptyList() // Ikke veileder
-            } else {
-                listOf(AzureAdGruppe("0000-MOCK-sosialhjelp-dialog-veileder", "Veiledere"))
-            }
+            val grupper = hentGrupper(id)
             ResponseEntity.ok(AzureAdGrupper(value = grupper))
         } catch (e: MockAltException) {
             log.info("Feil ved henting av brukers grupper: ${e.message}")
             ResponseEntity.status(HttpStatus.UNAUTHORIZED).build()
         }
     }
 
+    private fun hentGrupper(id: String): List<AzureAdGruppe> {
```

Comment:
evt  mock-grupper for modia kommer kanskje etter hvert?

Reply:
Jeg gjennbruker dialog sin gruppe. Vi bør kanskje lage et GUI for det på sikt.